### PR TITLE
[HLRC] Add GetRollupIndexCaps API

### PR DESCRIPTION
**Pull request by polyfractal**
Nearly identical to the GetRollupCaps API since the response is basically the same format, just different content.  The request object is slightly different too since it accepts a list of indices/index patterns and IndicesOptions

Also refactors the caps response tests a bit to share the same abstract class to reduce duplication of test code

Related to https://github.com/elastic/elasticsearch/issues/29827

### CODE REVIEW

File: client/rest-high-level/src/main/java/org/elasticsearch/client/rollup/GetRollupIndexCapsResponse.java [hunk 1]
```diff
@@ -0,0 +1,92 @@
...
+import org.elasticsearch.common.xcontent.XContentBuilder;
+import org.elasticsearch.common.xcontent.XContentParser;
+
+import java.io.IOException;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Objects;
+
+public class GetRollupIndexCapsResponse implements ToXContentObject {
```
Comment (hub-cap) [inline client/rest-high-level/src/main/java/org/elasticsearch/client/rollup/GetRollupIndexCapsResponse.java]: Generally speaking these responses shouldnt be `ToXContent` classes. I assume this is for a test case below. We have some newer HLRC's that have a diff test, see TermVectorsResponseTests. It looks like you picked the short straw that came after we modified the tests, sorry :)

**Review by hub-cap — CHANGES_REQUESTED** [review]
Looks good generally. Ill inspect it a bit more once youve modified the tests so they use the `org.elasticsearch.test.AbstractXContentTestCase.xContentTester`

File: client/rest-high-level/src/main/java/org/elasticsearch/client/rollup/GetRollupIndexCapsResponse.java (hunk 1 above)
Comment (hub-cap) [reply client/rest-high-level/src/main/java/org/elasticsearch/client/rollup/GetRollupIndexCapsResponse.java]: no, we are just changing things quickly on the client :)

**Review by hub-cap — APPROVED** [review]
great work updating the test. sorry it took a while to update.

**Comment by polyfractal** [comment]
Thanks @hub-cap! <3

**Pull request merged**